src/libstd/primitive_docs.rs

```diff
@@ -722,3 +722,104 @@ mod prim_isize { }
 ///
 #[stable(feature = "rust1", since = "1.0.0")]
 mod prim_usize { }
+
+#[doc(primitive = "fn")]
+//
+/// Function pointers, like `fn(usize) -> bool`.
+///
+/// *See also the traits [`Fn`], [`FnMut`], and [`FnOnce`].*
+///
+/// [`Fn`]: ops/trait.Fn.html
+/// [`FnMut`]: ops/trait.FnMut.html
+/// [`FnOnce`]: ops/trait.FnOnce.html
+///
+/// Plain function pointers are obtained by casting either plain functions, or closures that don't
+/// capture an environment:
+///
+/// ```
+/// fn add_one(x: usize) -> usize {
+///     x + 1
+/// }
+///
+/// let ptr: fn(usize) -> usize = add_one;
+/// assert_eq!(ptr(5), 6);
+///
+/// let clos: fn(usize) -> usize = |x| x + 5;
+/// assert_eq!(clos(5), 10);
+/// ```
+///
+/// In addition to varying based on their signature, function pointers come in two flavors: safe
+/// and unsafe. Plain `fn()` function pointers can only point to safe functions,
+/// while `unsafe fn()` function pointers can point to safe or unsafe functions.
+///
+/// ```
+/// fn add_one(x: usize) -> usize {
+///     x + 1
+/// }
+///
+/// unsafe fn add_one_unsafely(x: usize) -> usize {
+///     x + 1
+/// }
+///
+/// let safe_ptr: fn(usize) -> usize = add_one;
+///
+/// //ERROR: mismatched types: expected normal fn, found unsafe fn
+/// //let bad_ptr: fn(usize) -> usize = add_one_unsafely;
+///
+/// let unsafe_ptr: unsafe fn(usize) -> usize = add_one_unsafely;
+/// let really_safe_ptr: unsafe fn(usize) -> usize = add_one;
+/// ```
+///
+/// On top of that, function pointers can vary based on whether they're external, and then by what
+/// ABI they use. For example, `fn()` is different from `extern "C" fn()`, which itself is
+/// different from `extern "stdcall" fn()`, and so on for the various ABIs that Rust supports.
+/// Non-`extern` functions have an ABI of `"Rust"`, and `extern` functions without an explicit ABI
+/// have an ABI of `"C"`. For more information, see [the nomicon's section on foreign calling
+/// conventions][nomicon-abi].
+///
+/// [nomicon-abi]: ../nomicon/ffi.html#foreign-calling-conventions
+///
+/// Extern function declarations can also be *variadic*, allowing them to be called with a variable
+/// number of arguments. Normal rust functions, even those with an `extern "ABI"`, cannot be
+/// variadic. For more information, see [the nomicon's section on variadic
```

Comment:
> Extern function declarations can also be *variadic*, allowing them to be called with a variable number of arguments. Normal rust functions, even those with an `extern "ABI"`, cannot be variadic.

It's only functions with the `"C"` or `"cdecl"` that can be variadic so I think it would be clearer to say just something like ``Functions with the `"C"` or `"cdecl"` ABI can also be *variadic*, allowing them to be called with a variable number of arguments.``.